src/shared/components/ui/Pagination/index.tsx

```diff
@@ -0,0 +1,188 @@
+'use client';
+
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { useEffect } from 'react';
+
+import { cn } from '@/shared/libs/cn';
+
+/**
+ * Pagination 컴포넌트에 전달되는 props 타입 정의
+ *
+ * @property currentPage - 현재 페이지 번호
+ * @property totalPages - 전체 페이지 수
+ * @property onPageChange - 페이지 변경 시 호출되는 콜백 함수
+ * @property pageRange - 보여줄 페이지 버튼의 수 (기본값: 5)
+ * @property className - 페이지 버튼 추가적인 커스터마이징 스타일을 위한 클래스 이름
+ */
+type PaginationProps = {
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+  pageRange?: number; // default: 5
+  className?: string;
+};
```

Comment:
이 부분은 저희 컨벤션에 따라 `interface`로 수정해주시는 걸 권장드립니다!

```suggestion
interface PaginationProps {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
  pageRange?: number; // default: 5
  className?: string;
};
```